programmers/모의고사.js

```diff
@@ -0,0 +1,46 @@
+function solution(answers) {
+    const answerArr = []
+    const supo1 = [1,2,3,4,5];
+    const supo2 = [2,1,2,3,2,4,2,5];
+    const supo3 = [3,3,1,1,2,2,4,4,5,5];
+    const rank = [0,0,0]; //result[0] -> supo1
+    let i = 0; 
+    let j = 0; 
+    let k = 0;
+    
+    for (let answer of answers) {
+        if (answer === supo1[i]) {
+            rank[0]++;
+            i++;
+            if (i === 5) i = 0;
+        }
+        else {
+            i++;
+            if (i === 5) i = 0; 
+        }
+        
+        if (answer === supo2[j]) {
+            rank[1]++;
+            j++;
+            if (j === 8) j = 0;
+        }
+        else {
+            j++;
+            if (j === 8) j = 0;
+        }
+        
+        if (answer === supo3[k]) {
+            rank[2]++;
+            k++;
+            if (k === 10) k = 0;
+        }
+        else {
+            k++;
+            if (k === 10) k = 0;
+        }        
+    }    
```

Comment:
이게 최선인가요?
이런 너저분한 코드는 협업에 방해가 됩니다.

남들이 한 코드를 한번 살펴보시고 좀더 수학적인 접근이 필요합니다.
예를 들어, 김지수님의 코드를 한 번 살펴보시면 도움이 될겁니다.
```suggestion
    for (let answer of answers) {
        if (answer === supo1[i]) {
            rank[0]++;
            i++;
            if (i === 5) i = 0;
        }
        else {
            i++;
            if (i === 5) i = 0; 
        }
        
        if (answer === supo2[j]) {
            rank[1]++;
            j++;
            if (j === 8) j = 0;
        }
        else {
            j++;
            if (j === 8) j = 0;
        }
        
        if (answer === supo3[k]) {
            rank[2]++;
            k++;
            if (k === 10) k = 0;
        }
        else {
            k++;
            if (k === 10) k = 0;
        }        
    }    
```